packages/ember-extension-support/tests/data_adapter_test.js

```diff
@@ -8,223 +8,233 @@ import {
 import { Object as EmberObject, A as emberA } from 'ember-runtime';
 import EmberDataAdapter from '../data_adapter';
 import { Application as EmberApplication, Resolver as DefaultResolver } from 'ember-application';
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
+import { ModuleBasedResolver } from 'internal-test-helpers/test-resolver';
+
 
 let adapter, App;
 const Model = EmberObject.extend();
 
+const PostClass = Model.extend();
+
 const DataAdapter = EmberDataAdapter.extend({
   detect(klass) {
     return klass !== Model && Model.detect(klass);
-  }
-});
-
-QUnit.module('Data Adapter', {
-  setup() {
-    run(() => {
-      App = EmberApplication.create();
-      App.toString = function() { return 'App'; };
-      App.deferReadiness();
-      App.register('data-adapter:main', DataAdapter);
-    });
   },
-  teardown() {
-    run(() => {
-      adapter.destroy();
-      App.destroy();
+  init() {
+    this._super(...arguments);
+    let self = this;
+    this.set('containerDebugAdapter', {
+      canCatalogEntriesByType(type) {
+        return true;
+      },
+      catalogEntriesByType(type) {
+        return emberA(['post']);
+      }
     });
   }
 });
 
-QUnit.test('Model types added with DefaultResolver', function() {
-  App.Post = Model.extend();
-
-  adapter = App.__container__.lookup('data-adapter:main');
-  adapter.reopen({
-    getRecords() {
-      return emberA([1, 2, 3]);
-    },
-    columnsForType() {
-      return [{ name: 'title', desc: 'Title' }];
-    }
-  });
-
-  run(App, 'advanceReadiness');
-
-  function modelTypesAdded(types) {
-    equal(types.length, 1);
-    let postType = types[0];
-    equal(postType.name, 'post', 'Correctly sets the name');
-    equal(postType.count, 3, 'Correctly sets the record count');
-    strictEqual(postType.object, App.Post, 'Correctly sets the object');
-    deepEqual(postType.columns, [{ name: 'title', desc: 'Title' }], 'Correctly sets the columns');
-  }
 
-  run(adapter, 'watchModelTypes', modelTypesAdded);
-});
+moduleFor('Data Adapter', class extends ApplicationTestCase {
 
-QUnit.test('getRecords gets a model name as second argument', function() {
-  App.Post = Model.extend();
+  ['@test Model types added with DefaultResolver']() {
```

Comment:
So this test is specifically mentioning `DefaultResolver`. By migrating to the `ApplicationTestCase` and using `add` etc, it is no longer using the default resolver. Instead it now uses the TestResolver.

Either the test description is wrong and the data adapter doesn't require the DefaultResolver, or this test should not be ported. I''m not sure which.